In [7]:
'''
    This main function retrieves documents from a file and writes them to a Mongo database collection,
        where each line of the file is a document represented as a json item
    It assumes that the Mongo DB server is running.
    Command line usage:
        python DB_from_file.py <DB name> <collection name> <filepath>
    Example:      python DB_from_file.py fbusers cdelta out/fb-delta-feed.txt 
        (where quotes may not be necessary)
'''

# import json
# import sys
# from DB_fn import save_to_DB

# # use a main so can get command line arguments
# if __name__ == '__main__':
#     # Make a list of command line arguments, omitting the [0] element
#     # which is the script itself.
#     args = sys.argv[1:]
#     if not args or len(args) < 3:
#         print('usage: python DB_from_file.py <DB name> <collection name> <filepath>')
#         sys.exit(1)
#     DBname = args[0]
#     DBcollection = args[1]
#     filename = args[2]  
    
#     # get the data from the file
#     fload = open(filename, encoding='utf-8')
#     resultString = fload.read()
#     # loads converts from a string back to a json structure
#     doclist = json.loads(resultString)  
#     print ("Read", len(doclist), "from file:", filename)

#     # in case this was saved from a database, we delete the database id key _id, so that DB will assign unique key
#     for doc in doclist:
#         if '_id' in doc.keys():
#             del doc['_id']
    
#     # use the save and load functions in this program
#     save_to_DB(DBname, DBcollection, doclist)



'\n    This main function retrieves documents from a file and writes them to a Mongo database collection,\n        where each line of the file is a document represented as a json item\n    It assumes that the Mongo DB server is running.\n    Command line usage:\n        python DB_from_file.py <DB name> <collection name> <filepath>\n    Example:      python DB_from_file.py fbusers cdelta out/fb-delta-feed.txt \n        (where quotes may not be necessary)\n'

## Terminal commands to get files to read to MONGODB


Terminal Text in order to get the files to read to mongodb:

Last login: Thu Aug 18 19:30:33 on ttys001
(base) pergolicious@pergolicious ~ % cd /Users/pergolicious/Downloads/         
(base) pergolicious@pergolicious Downloads % python DB_from_file.py bball bbcoll bball_mm2.txt
Traceback (most recent call last):
  File "/Users/pergolicious/Downloads/DB_from_file.py", line 13, in <module>
    from DB_fn import save_to_DB
ModuleNotFoundError: No module named 'DB_fn'
(base) pergolicious@pergolicious Downloads % python db_from_file.py bball bbcoll bball_mm2.txt
Read 2000 from file: bball_mm2.txt
Saved 2000 documents to DB collection bball bbcoll
(base) pergolicious@pergolicious Downloads % 

In [2]:
import pymongo

client = pymongo.MongoClient('localhost',27017) 

#list the databases defined 
client.list_database_names()
db = client.bball
db.list_collection_names()
coll=db.bbcoll
docs = coll.find()

In [3]:
#convert the document cursor to a list of documents 
>>> doclist = [tweet for tweet in docs] 
>>> len(doclist)




2000

In [4]:
# show difference from print 
print(doclist[:1])
# Look through this to point out field names

[{'_id': ObjectId('62fedc9ce4e1ed9e84615062'), 'user': {'profile_background_tile': True, 'friends_count': 217, 'profile_sidebar_fill_color': 'EFEFEF', 'id_str': '513196438', 'is_translation_enabled': False, 'profile_link_color': 'F70808', 'followers_count': 256, 'location': '', 'protected': False, 'default_profile_image': False, 'contributors_enabled': False, 'favourites_count': 2187, 'profile_background_color': 'BFD0D9', 'statuses_count': 1104, 'id': 513196438, 'profile_banner_url': 'https://pbs.twimg.com/profile_banners/513196438/1437359097', 'created_at': 'Sat Mar 03 13:54:37 +0000 2012', 'profile_image_url_https': 'https://pbs.twimg.com/profile_images/732514428051181568/OkOVa8Ia_normal.jpg', 'time_zone': 'Eastern Time (US & Canada)', 'follow_request_sent': None, 'listed_count': 1, 'utc_offset': -14400, 'lang': 'en', 'is_translator': False, 'name': 'Will', 'description': "Do not worry about tomorrow, for tomorrow will worry about itself.-Matthew 6:34 HSC'20", 'profile_use_background

In [5]:
def print_tweet_data(tweets):
    for tweet in tweets:
        print('\nDate:', tweet['created_at'])
        print('From:', tweet['user']['name'])
        print('Message', tweet['text'])

In [6]:
print_tweet_data(doclist[:5])


Date: Mon Mar 27 22:30:30 +0000 2017
From: Will
Message RT @HowardWKYT: The final seconds of the Kentucky-North Carolina game was an emotional roller coaster for Big Blue fans. #marchmadness #WKY…

Date: Mon Mar 27 22:30:22 +0000 2017
From: Kahlen Donatell
Message RT @WhistleSports: When you perfectly time the #UNC buzzer beater 😱🏀 #MarchMadness

(via:@SamuelGrubbs1 ) https://t.co/Ol2ibpZjB4

Date: Mon Mar 27 22:30:21 +0000 2017
From: Jesús
Message RT @HowardWKYT: The final seconds of the Kentucky-North Carolina game was an emotional roller coaster for Big Blue fans. #marchmadness #WKY…

Date: Mon Mar 27 22:30:14 +0000 2017
From: plug,
Message RT @BleacherReport: And then there were four... #MarchMadness https://t.co/0MbxpgAuUC

Date: Mon Mar 27 22:30:11 +0000 2017
From: Joni Dickerson
Message RT @mycarolinastdnt: RT if you'll be cheering on @GamecockWBB and @dawnstaley tonight. Let's Go Gamecocks! #MarchMadness


Terminal Text in order to get the files to read to mongodb:

Last login: Thu Aug 18 19:30:33 on ttys001
(base) pergolicious@pergolicious ~ % cd /Users/pergolicious/Downloads/         
(base) pergolicious@pergolicious Downloads % python DB_from_file.py bball bbcoll bball_mm2.txt
Traceback (most recent call last):
  File "/Users/pergolicious/Downloads/DB_from_file.py", line 13, in <module>
    from DB_fn import save_to_DB
ModuleNotFoundError: No module named 'DB_fn'
(base) pergolicious@pergolicious Downloads % python db_from_file.py bball bbcoll bball_mm2.txt
Read 2000 from file: bball_mm2.txt
Saved 2000 documents to DB collection bball bbcoll
(base) pergolicious@pergolicious Downloads % 

In [ ]:
# Change directory in terminal to where the files are downloaded:

    # cd /Users/pergolicious/Downloads/  

# Insert the following line (may have to change names of files to ensure they match code):
    # python db_from_file.py bball bbcoll bball_mm2.txt
    


In [8]:
for tweet in doclist[:5]:
    print(tweet['user']['screen_name'])

frenchythe1st
KahDon
_jonwilliams
21shawtyy
jonidickerson


In [9]:
# solution from class avtivity:

Reading Tweets Activity Solution
Possible responses to the activity
>>> import pymongo
>>> client = pymongo.MongoClient('localhost',27017) >>> client.database_names()
['admin', 'bball', 'fbusers', 'lax', 'local', 'peopledb', 'usgs'] >>> db = client.bball
>>> db.collection_names()
['bbcoll']
>>> coll = db.bbcoll
>>> docs = coll.find()
>>> doclist = [tweet for tweet in docs]
>>> len(doclist)
2000
>>> print(doclist[:1])
[{'id': 846489633387089920, 'user': {'contributors_enabled': False, 'protected': False, 'profile_background_color': 'BFD0D9', 'geo_enabled': True, 'profile_sidebar_fill_color': 'EFEFEF', 'profile_image_url_https': 'https://pbs.twimg.com/profile_images/732514428051181568/OkOVa8Ia_normal.jpg', 'profile_banner_url': 'https://pbs.twimg.com/profile_banners/513196438/1437359097', 'lang': 'en', 'url': None, 'following': None, 'followers_count': 256, 'id': 513196438, 'has_extended_profile': True, 'location': '', 'profile_sidebar_border_color': 'FFFFFF', 'friends_count': 217, 'listed_count': 1, 'profile_background_image_url': 'http://pbs.twimg.com/profile_background_images/850934042/26e0a43c1f821ac098571fb3de80944c.j peg', 'is_translation_enabled': False, 'profile_background_image_url_https': 'https://pbs.twimg.com/profile_background_images/850934042/26e0a43c1f821ac098571fb3de80944c. jpeg', 'favourites_count': 2187, 'name': 'Will', 'entities': {'description': {'urls': []}}, 'profile_link_color': 'F70808', 'profile_text_color': '333333', 'default_profile_image': False, 'screen_name': 'frenchythe1st', 'translator_type': 'none', 'created_at': 'Sat Mar 03 13:54:37 +0000 2012', 'profile_image_url': 'http://pbs.twimg.com/profile_images/732514428051181568/OkOVa8Ia_normal.jpg', 'profile_use_background_image': True, 'statuses_count': 1104, 'description': "Do not worry about tomorrow, for tomorrow will worry about itself.-Matthew 6:34 HSC'20", 'profile_background_tile': True, 'is_translator': False, 'default_profile': False, 'verified': False, 'time_zone': 'Eastern Time (US & Canada)', 'follow_request_sent': None, 'id_str': '513196438', 'utc_offset': -14400, 'notifications': None},
'coordinates': None, 'in_reply_to_status_id_str': None, 'retweeted_status': {'id': 846214230026178564, 'user': {'contributors_enabled': False, 'protected': False, 'profile_background_color': '000000', 'geo_enabled': True, 'profile_sidebar_fill_color': 'C0DFEC', 'profile_image_url_https': 'https://pbs.twimg.com/profile_images/575376840304386049/atONJG3G_normal.jpeg', 'profile_banner_url': 'https://pbs.twimg.com/profile_banners/27059989/1431999643', 'lang': 'en', 'url': 'http://t.co/C2UFfLkjB1', 'following': None, 'followers_count': 3330, 'id': 27059989, 'has_extended_profile': False, 'location': 'Lexington, KY', 'profile_sidebar_border_color': 'A8C7F7', 'friends_count': 573, 'listed_count': 77, 'profile_background_image_url': 'http://pbs.twimg.com/profile_background_images/596008542/6qfqqg0jndpp3su53ay5.jpeg', 'is_translation_enabled': False, 'profile_background_image_url_https': 'https://pbs.twimg.com/profile_background_images/596008542/6qfqqg0jndpp3su53ay5.jpeg', 'favourites_count': 1231, 'name': 'Lee K. Howard', 'entities': {'description': {'urls': []}, 'url': {'urls': [{'display_url': 'facebook.com/profile.php?id...', 'expanded_url': 'http://www.facebook.com/profile.php?id=100003241678454', 'indices': [0, 22], 'url': 'http://t.co/C2UFfLkjB1'}]}}, 'profile_link_color': '0084B4', 'profile_text_color': '333333', 'default_profile_image': False, 'screen_name': 'HowardWKYT', 'translator_type': 'none', 'created_at': 'Fri Mar 27 18:09:16 +0000 2009', 'profile_image_url': 'http://pbs.twimg.com/profile_images/575376840304386049/atONJG3G_normal.jpeg', 'profile_use_background_image': True, 'statuses_count': 9463, 'description': 'Sports Anchor/Reporter for CBS/FOX in Lexington Kentucky, providing sports news and my random rants and chants!', 'profile_background_tile': False, 'is_translator': False, 'default_profile': False, 'verified': True, 'time_zone': 'Eastern Time (US & Canada)', 'follow_request_sent': None, 'id_str': '27059989', 'utc_offset': -14400, 'notifications': None}, 'coordinates': None, 'in_reply_to_status_id_str': None, 'metadata': {'iso_language_code': 'en', 'result_type': 'recent'}, 'in_reply_to_user_id': None, 'lang': 'en', 'favorite_count': 12590, 'is_quote_status': False, 'geo': None, 'possibly_sensitive': False, 'in_reply_to_status_id': None, 'favorited': False, 'in_reply_to_screen_name': None, 'created_at': 'Mon Mar 27 04:16:08 +0000 2017', 'in_reply_to_user_id_str': None, 'entities': {'hashtags': [], 'urls': [{'display_url': 'twitter.com/i/web/status/8...', 'expanded_url': 'https://twitter.com/i/web/status/846214230026178564', 'indices': [106, 129], 'url': 'https://t.co/TPZ6PuXHxH'}], 'user_mentions': [], 'symbols': []}, 'truncated': True, 'source': '<a href="http://twitter.com" rel="nofollow">Twitter Web Client</a>', 'retweet_count': 9739, 'contributors': None, 'place': {'contained_within': [], 'country_code': 'US', 'id': '6ffcf3b0b904bbcb', 'country': 'United States', 'full_name': 'Kentucky, USA', 'url': 'https://api.twitter.com/1.1/geo/id/6ffcf3b0b904bbcb.json', 'place_type': 'admin', 'attributes': {}, 'name': 'Kentucky', 'bounding_box': {'type': 'Polygon', 'coordinates': [[[-89.57151, 36.497129], [- 81.964971, 36.497129], [-81.964971, 39.147359], [-89.57151, 39.147359]]]}}, 'retweeted': False, 'text': 'The final seconds of the Kentucky-North Carolina game was an emotional roller coaster for Big Blue fans.... https://t.co/TPZ6PuXHxH', 'id_str': '846214230026178564'}, 'metadata': {'iso_language_code': 'en', 'result_type': 'recent'}, 'in_reply_to_user_id': None, 'lang': 'en', 'favorite_count': 0, '_id': ObjectId('5aca8b348f5b161128bf4496'), 'is_quote_status': False, 'geo': None, 'in_reply_to_status_id': None, 'favorited': False, 'in_reply_to_screen_name': None, 'created_at': 'Mon Mar 27 22:30:30 +0000 2017', 'in_reply_to_user_id_str': None, 'entities': {'hashtags': [{'text': 'marchmadness', 'indices': [121, 134]}], 'urls': [], 'user_mentions': [{'id': 27059989, 'id_str': '27059989', 'indices': [3, 14], 'screen_name': 'HowardWKYT', 'name': 'Lee K. Howard'}], 'symbols': []}, 'truncated': False, 'contributors': None,

'retweet_count': 9739, 'place': None, 'retweeted': False, 'text': 'RT @HowardWKYT: The final seconds of the Kentucky-North Carolina game was an emotional roller coaster for Big Blue fans. #marchmadness #WKY...', 'id_str': '846489633387089920', 'source': '<a href="http://twitter.com/download/iphone" rel="nofollow">Twitter for iPhone</a>'}]
>>> def print_tweet_data(tweets):
...
...
...
...
...
>>> print_tweet_data(doclist[:1])
for tweet in tweets:
print('\nDate: ',tweet['created_at']) print('From: ',tweet['user']['name']) print('Message: ',tweet['text'])
Date: Mon Mar 27 22:30:30 +0000 2017 From: Will
Message: RT @HowardWKYT: The final seconds of the Kentucky-North Carolina game was an emotional roller coaster for Big Blue fans. #marchmadness #WKY...
>>> for tweet in doclist[:5]:
... print(tweet['user']['screen_name']) ...
frenchythe1st
KahDon
_jonwilliams
21shawtyy
jonidickerson